### PR TITLE
 TYP: Add MyPy Error Codes

### LINKED ISSUE #29197: MyPy Error Codes
mypy 0.73 came with support for error codes:

http://mypy-lang.blogspot.com/2019/09/mypy-730-released.html

I think we should add these to CI and update docs to require these going forward for any `type: ignore` statements

@simonjayhawkins

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] closes #29197
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@WillAyd have added a code-check to enforce this but may not be able to enable until we bump mypy due a bug in mypy not suppressing some additional notes, so have just commented out the code check for now

File: environment.yml [hunk 1]
```diff
@@ -109,3 +109,4 @@ dependencies:
   - pip:
     - git+https://github.com/pandas-dev/pydata-sphinx-theme.git@master
     - git+https://github.com/numpy/numpydoc
+    - pyflakes>=2.2.0
```
Comment (WillAyd) [inline environment.yml:112]: I think this is available from conda?

File: pandas/io/common.py [hunk 2]
```diff
@@ -121,7 +121,15 @@ def stringify_path(
     """
     if hasattr(filepath_or_buffer, "__fspath__"):
         # https://github.com/python/mypy/issues/1424
-        return filepath_or_buffer.__fspath__()  # type: ignore
+        # error: Item "str" of "Union[str, Path, IO[str]]" has no attribute
```
Comment (WillAyd) [inline pandas/io/common.py:124]: What criteria are you using to include a comment or not?

**Review by WillAyd — COMMENTED** [review]
Cool! Yea I think being more specific is always worthwhile

File: environment.yml (hunk 1 above)
Comment (simonjayhawkins) [reply environment.yml:112]: I'll give it a go on the CI, IIRC I had issues locally with conda but later found an issue in the environment. conda list showed 2.2.0 but flake8 was still using 2.1.1

File: pandas/io/common.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/io/common.py:124]: Hopefully, always. it makes it much easier when browsing the code for type: ignores to remove. e.g. #34956 

While adding the error messages here, it was immediately apparent that those ignores we masking another issue. If this was already merged to master, the error message would be in the diff (as deleted) for #35344 and make the intent of the changes more clear and hence easier to review imo.

File: environment.yml (hunk 1 above)
Comment (simonjayhawkins) [reply environment.yml]: conda doesn't like it.

**Comment by simonjayhawkins** [comment]
@WillAyd gentle ping!

**Review by WillAyd — APPROVED** [review]
lgtm. maybe give it a day to see if @jreback has any feedback otherwise can merge

**Comment by WillAyd** [comment]
Thanks @simonjayhawkins

**Pull request merged**